deeprank/learn/DataSet.py

```diff
@@ -418,15 +418,19 @@ def create_index_molecules(self):
                 fh5 = h5py.File(fdata, 'r')
                 mol_names = list(fh5.keys())
                 mol_names = self._select_pdb(mol_names)
-                for k in mol_names:
-                    if self.filter(fh5[k]):
-                        self.index_complexes += [(fdata,
-                                                  k, None, None)]
-                        for irot in range(self.data_augmentation):
-                            axis, angle = pdb2sql.transform.get_rot_axis_angle(
-                                self.rotation_seed)
-                            self.index_complexes += [
-                                (fdata, k, angle, axis)]
+                # to speed up in case of no filtering:
+                if not self.dict_filter:
+                    self.index_complexes = [[fdata, k, None, None] for k in mol_names]
+                else:
+                    for k in mol_names:            
+                        if self.filter(fh5[k]):
```

Comment:
The `if self.filter(fh5[k]):` is a call to a function that checks each datapoint if it passes a certain filter, otherwise it is not added to the `self.index_complexes`. So probably the only way to significantly speed up this process (if `dict_filter` is required) is to parallelize this with multiple cpus.